app/main.py

```diff
@@ -1,2 +1,89 @@
+from dataclasses import dataclass
+from typing import Hashable, Any
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, initial_capacity: int = 8) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.load_factor = 2 / 3
+        self.hash_table: list = [None] * self.capacity
+
+    def calculate_index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+
+        while (
+                self.hash_table[index] is not None
+                and self.hash_table[index].key != key
+        ):
+            index = (index + 1) % self.capacity
+
+        return index
+
+    @property
+    def current_max_size(self, threshold: float = 2 / 3) -> float:
+        return self.capacity * threshold
+
+    def resize(self, capacity_multiplier: int = 2) -> None:
+        previous_hash_table = self.hash_table
+
+        self.__init__(self.capacity * capacity_multiplier)
+
+        for node in previous_hash_table:
+            if node is not None:
+                self.__setitem__(node.key, node.value)
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self.calculate_index(key)
+
+        if self.hash_table[index] is None:
+            if self.size + 1 >= self.current_max_size:
+                self.resize()
+                self.__setitem__(key, value)
```

Comment:
In the `__setitem__` method, calling `self.__setitem__(key, value)` after resizing can lead to infinite recursion if the resize does not resolve the collision. Consider restructuring the logic to avoid this potential issue.